exporter/datadogexporter/internal/logs/sender.go

```diff
@@ -64,30 +65,62 @@ func (s *Sender) SubmitLogs(ctx context.Context, payload []datadogV2.HTTPLogItem
 		s.logger.Debug("Submitting logs", zap.Any("payload", payload))
 	}
 
-	// enable sending gzip
-	// Get a fresh opts for each request to avoid duplicating ddtags
-	opts := *datadogV2.NewSubmitLogOptionalParameters().WithContentEncoding(datadogV2.CONTENTENCODING_GZIP)
+	var prevDdtags string
+	var logItemBatch []datadogV2.HTTPLogItem
+
+	if len(payload) > 0 && payload[0].HasDdtags() {
+		prevDdtags = *payload[0].Ddtags
+	}
 
 	// Correctly sets apiSubmitLogRequest ddtags field based on tags from translator Transform method
-	if payload[0].HasDdtags() {
-		tags := datadog.PtrString(payload[0].GetDdtags())
-		if opts.Ddtags != nil {
-			tags = datadog.PtrString(fmt.Sprint(*opts.Ddtags, ",", payload[0].GetDdtags()))
+	for i, logItem := range payload {
+		// enable sending gzip
+		// Get a fresh opts for each request to avoid duplicating ddtags
+		opts := *datadogV2.NewSubmitLogOptionalParameters().WithContentEncoding(datadogV2.CONTENTENCODING_GZIP)
+
+		var tags string
+		if logItem.HasDdtags() {
+			tags = logItem.GetDdtags()
+			if opts.Ddtags != nil {
+				tags = fmt.Sprint(*opts.Ddtags, ",", logItem.GetDdtags())
+			}
+			opts.Ddtags = datadog.PtrString(tags)
 		}
-		opts.Ddtags = tags
-	}
-	_, r, err := s.api.SubmitLog(ctx, payload, opts)
-	if err != nil {
-		if r != nil {
-			b := make([]byte, 1024) // 1KB message max
-			n, _ := r.Body.Read(b)  // ignore any error
-			s.logger.Error("Failed to send logs", zap.Error(err), zap.String("msg", string(b[:n])), zap.String("status_code", r.Status))
-			return err
+
+		// Batches consecutive log items with the same tags to be submitted together
```

Comment:
Please put this line inside the `if` statement, as per our [coding guidelines](https://datadoghq.atlassian.net/wiki/spaces/OTEL/pages/2580119947/Go+Coding+Guidelines).

Reply:
You did not address this one.